internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
+}
```

Comment:
The use of a generic function `GenerateSchema[T any]()` introduces complexity without apparent benefit, given that the only use case is for `ReviewSuggestionsResponse`. If the schema generation is specific to `ReviewSuggestionsResponse`, consider removing the generic type parameter and directly implementing the function for this type to simplify the code.